framework/include/utils/MathUtils.h

```diff
@@ -132,7 +132,7 @@ poly(std::vector<Real> c, const T x, const bool derivative)
 
 template <typename T>
 T
-clamp(T x, Real lowerlimit, Real upperlimit)
+clamp(const T & x, Real lowerlimit, Real upperlimit)
```

Comment:
So if the purpose of this function will always be to clamp a variable, you should make this a void function and pass in a non-const reference that you modify in the function. Making this a const reference, but then using this function to assign the value of the original variable seems error-prone.

Reply:
Actually on second thought, I prefer having it return something, as that's what all "math" functions do.

Reply:
Yeah, I thought about making it a void, but I was thinking it may be worth while to make it easier to retain a clamped and unclamped version, or be able to use it in an if statement...either way works well, probably makes the most sense to have it consistent with all other math like you said